src/odemis/model/_futures.py

```diff
@@ -538,8 +546,20 @@ def _on_future_done(self, f):
             self.set_result(None)
 
     def _estimate_end(self):
-        start, end = self.get_progress()
-        return start + sum(self.futures[f] for f in self.futures if not f.done())
+        """
+        Estimate the end of the batch future based on the current time plus the
+        estimated time for all futures that still need to be executed.
+
+        returns: (float) Newly estimated end time based for the batch future.
+        """
+        duration = time.time()
+        for f, t in self.futures.items():
+            # only take futures into account that are not executed (finished) yet
+            if not f.done():
+                duration += t
+        # FIXME: it does not update often enough in the GUI, the estimate seems reasonable, but
+        #  gauge does not execute 1s when showing 1 second has passed -> takes longer thought time would match well
```

Comment:
I don't get the issue that you're trying to describe in this comment.
Be aware that set_progress() is maybe not called so frequently, so it's not going to change the GUI directly. When the GUI doesn't receive any update on the estimated time, it assumes that the previous estimation was correct and update the displayed remaining time every 1s.

Reply:
It was not updated every 1s put stayed at the same value for over 10 sec.

Reply:
For imaging of 2 overview images I see the following:
estimated time:  ~ 3 min 45sec (dwell time =  25ns)
start time acq: ~ 2 min 23sec updates once to 2 min 22 sec, then stays there over 10 sec before counting down
sec image: jumps to ~36sec

Reply:
Found that this issue is related to the bad estimated time for the tiled acquisition and has nothing to do with the changes here.